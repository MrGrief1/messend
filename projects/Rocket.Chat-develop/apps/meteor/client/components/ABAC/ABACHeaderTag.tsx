import type { IRoom } from '@rocket.chat/core-typings';
import { Box, Palette } from '@rocket.chat/fuselage';
import { useTranslation } from 'react-i18next';

import { HeaderTag } from '../Header';

type ABACHeaderTagProps = {
	room: IRoom;
};

const ABACHeaderTag = ({ room }: ABACHeaderTagProps) => {
	const { t } = useTranslation();

	// @ts-expect-error to be implemented
	if (!room.abacAttributes) {
		return null;
	}

	return (
		<HeaderTag title={t('ABAC_header_tag_title')}>
			<Box color={Palette.statusColor['status-font-on-warning'].toString()} fontWeight='700'>
				{t('ABAC_header_tag')}
			</Box>
		</HeaderTag>
	);
};

export default ABACHeaderTag;
