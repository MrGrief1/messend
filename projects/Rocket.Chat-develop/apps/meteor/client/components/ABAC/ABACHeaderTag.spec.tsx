import type { IRoom } from '@rocket.chat/apps-engine/definition/rooms';
import { mockAppRoot } from '@rocket.chat/mock-providers';
import { render, screen } from '@testing-library/react';
import { axe } from 'jest-axe';

import ABACHeaderTag from './ABACHeaderTag';

const appRoot = mockAppRoot()
	.withTranslations('en', 'core', {
		ABAC_header_tag_title: 'Only compliant users have access to attribute-based access controlled rooms.',
		ABAC_header_tag: 'ABAC',
	})
	.build();

const createMockRoom = (overrides: Partial<IRoom> = {}) => ({
	_id: 'room-id',
	t: 'c' as const,
	name: 'test-room',
	msgs: 0,
	u: { _id: 'user-id', username: 'testuser' },
	usersCount: 1,
	_updatedAt: new Date(),
	...overrides,
});

describe('ABACHeaderTag', () => {
	it('should render the ABAC tag when room has ABAC attributes', () => {
		const room = createMockRoom({
			// @ts-expect-error to be implemented
			abacAttributes: { someAttribute: 'value' },
		});

		const { baseElement } = render(<ABACHeaderTag room={room} />, { wrapper: appRoot });
		expect(baseElement).toMatchSnapshot();
	});

	it('should not render when room has no ABAC attributes', () => {
		const room = createMockRoom();

		render(<ABACHeaderTag room={room} />, { wrapper: appRoot });
		expect(screen.queryByText('ABAC')).not.toBeInTheDocument();
	});

	it('should have no accessibility violations when rendered', async () => {
		const room = createMockRoom({
			// @ts-expect-error to be implemented
			abacAttributes: { someAttribute: 'value' },
		});

		const { container } = render(<ABACHeaderTag room={room} />, { wrapper: appRoot });
		const results = await axe(container);
		expect(results).toHaveNoViolations();
	});

	it('should have no accessibility violations when not rendered', async () => {
		const room = createMockRoom();

		const { container } = render(<ABACHeaderTag room={room} />, { wrapper: appRoot });
		const results = await axe(container);
		expect(results).toHaveNoViolations();
	});
});
