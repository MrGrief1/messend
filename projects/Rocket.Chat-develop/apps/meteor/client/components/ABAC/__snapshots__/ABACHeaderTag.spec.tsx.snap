// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ABACHeaderTag should render the ABAC tag when room has ABAC attributes 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-1kopyx3"
    >
      <span
        class="rcx-box rcx-box--full rcx-tag rcx-tag--medium"
        title="Only compliant users have access to attribute-based access controlled rooms."
      >
        <span
          class="rcx-tag__inner"
        >
          <div
            class="rcx-box rcx-box--full rcx-css-12a8uky"
          >
            ABAC
          </div>
        </span>
      </span>
    </div>
  </div>
</body>
`;
